packages/next/build/webpack/loaders/eslint-loader/cache.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * Original Filesystem Cache implementation by babel-loader
+ * Licensed under the MIT License
+ *
+ * @see https://github.com/babel/babel-loader/commits/master/src/fs-cache.js
+ * @see https://github.com/babel/babel-loader/commits/master/src/cache.js
+ */
+
+/**
+ * Filesystem Cache
+ *
+ * Given a file and a transform function, cache the result into files
+ * or retrieve the previously cached files if the given file is already known.
+ *
+ * @see https://github.com/babel/babel-loader/issues/34
+ * @see https://github.com/babel/babel-loader/pull/41
+ */
+import fs from 'fs'
+import os from 'os'
+import { join } from 'path'
+import { promisify } from 'util'
+import zlib from 'zlib'
+import { createHash } from 'crypto'
+
+// @ts-ignore
+import findCacheDir from 'find-cache-dir'
+import { CLIEngine } from 'eslint'
+
+// Lazily instantiated when needed
+let defaultCacheDirectory: string | null = null
+
+const readFile = promisify(fs.readFile)
+const writeFile = promisify(fs.writeFile)
+const gunzip = promisify(zlib.gunzip)
+const gzip = promisify(zlib.gzip)
+
+const read = async (filename: string, compress: Boolean) => {
+  const data = await readFile(filename + (compress ? '.gz' : ''))
+  const content = (compress ? await gunzip(data) : data) as Buffer
+
+  return JSON.parse(content.toString())
+}
+
+const write = async (filename: string, compress: boolean, result: string) => {
+  const content = JSON.stringify(result)
+
+  const data = compress ? await gzip(content) : content
+  return writeFile(filename + (compress ? '.gz' : ''), data)
+}
+
+const filename = (source: string, identifier: string, options: any) => {
+  const hash = createHash('md4')
+
+  const contents = JSON.stringify({ source, options, identifier })
+
+  hash.update(contents)
+
+  return `${hash.digest('hex')}.json`
+}
+
+const handleCache = async (
+  directory: string,
+  params: any
+): Promise<CLIEngine.LintReport> => {
+  const {
+    source,
+    options = {},
+    transform,
+    cacheIdentifier,
+    cacheDirectory,
+    cacheCompression,
+  } = params
+
+  const file = join(directory, filename(source, cacheIdentifier, options))
```

Comment:
The cache needs to account for external `.eslintrc` config files changing right? 

It doesn't appear this is currently being done, this can be seen by running the `test/integration/eslint-loader/test/custom-config.test.js` tests and then deleting the `.eslintrc` for that test and re-running it (they still pass and they shouldn't).